rb/spec/integration/selenium/webdriver/bidi/script_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+require_relative '../spec_helper'
+
+module Selenium
+  module WebDriver
+    describe Script, only: {browser: %i[chrome edge firefox]} do
+      before { reset_driver!(web_socket_url: true) }
+      after(:all) { quit_driver }
+
+      it 'errors when bidi not enabled' do
+        reset_driver! do |driver|
+          expect {
+            driver.script
+          }.to raise_error(WebDriver::Error::WebDriverError, /Script methods requires enabling BiDi/)
+        end
+      end
+
+      it 'logs console messages' do
+        log_entries = []
+
+        driver.script.add_console_message_handler { |log| log_entries << log }
+        driver.navigate.to url_for('bidi/logEntryAdded.html')
+
+        driver.find_element(id: 'jsException').click
+        driver.find_element(id: 'consoleLog').click
+
+        wait.until { log_entries.any? }
+        expect(log_entries.size).to eq(1)
+        expect(log_entries.first).to include('level' => 'info',
```

Comment:
FYI There were some flaky failures related to Edge I fixed in 062c125c062eea270c9ce8dafcb8f92f32e1bf51. This happened because Edge was emitted "Driver started" console message by itself so I had to ensure we don't just wait for ANY console messages, but wait for specific "Hello, world!" one. You probably need to do the same in these specs too.

Reply:
This is a good point, but I do want us to be as specific as we can be for these. I think if we navigate before adding a handler then we should be safe.